BLSQ/iaso
# POLIO-1695 POLIO-1697 POLIO-1698 POLIO-1699 POLIO-1696 POLIO-1700 Vaccine module upload of pdf documents

Upload PDF files into the vaccine module for each type of file:

- VRF
- Pre Alert
- Form A
- Destruction Report
- Incident report

Related JIRA tickets : POLIO-1695, POLIO-1697, POLIO-1698, POLIO-1699, POLIO-1696, POLIO-1700

## Self proofreading checklist


## Changes

Add a custom storage manager to point to another public S3 Bucket if `AWS_PUBLIC_STORAGE_BUCKET_NAME` env var is set (otherwise it uses default storage)
Add "document" FileField on relevant models that uses this storage
Modification of the APIs to handle the files and uploads.
Needed a new Parser class to handle Nested Multipart Forms data which is supported by the standard but not by DRF out of the box.
Modify the frontend to handle these documents in all the screens for the relevant items (See above)

## How to test

Either directly via API but you would need to send data in Form Data (Not JSON) (See the code in the tests) or via the UI in all the Vaccine Stock and Supply Chain screens.

## Print screen / video


https://github.com/user-attachments/assets/0c8a7765-b625-4fd2-abe5-1dd615a0010a
